### ISSUE TITLE
Name does not uniquely identify Azure Web App

### ISSUE THREAD

**Issue by vanessalove** [OctopusDeploy/Issues#2554]
When an Azure account/subscription has more than 1 web app with the same name, the 'Deploy to Azure' step type does not properly distinguish between them.

Source: http://help.octopusdeploy.com/discussions/problems/46324

**Comment by octoreleasebot** [comment]
Release Note: Azure Web App step now supports specifying Resource Group

**Comment by BrianRichardson** [comment]
In my experience with v3.4.9, publishing an App Service hosted within App Service Environment, fails due to the incorrect publish profiles being retrieved.

I have raised this here too:
http://help.octopusdeploy.com/search?utf8=%E2%9C%93&q=resource+group&t=d&scope=all&category_id=40741

In my scenario, I have several App Service Environments, each with their own Resource Group.  Each ASE has an App Service with the same name and is distinguishable only by the ASE itself and/or the Resource Group.  For example:

ASE1 - ResourceGroup1
   WebAppA - ResourceGroup1
   WebAppB - ResourceGroup1

ASE2 - ResourceGroup2
   WebAppA - ResourceGroup2
   WebAppB - ResourceGroup2

Although the "Deploy an Azure Web App" step for "WebAppA" has Resource Group set to "ResourceGroup2", it incorrectly retrieves the Publish Profile for WebAppA within "ResourceGroup1".


Deployment log
==============

.....
Deploying to Azure WebApp 'WebApp(Staging)' in Resource Group WebApp-Test3, using subscription-id '### SubscriptionId ###'
September 19th 2016 12:45:00Verbose
Retrieving publishing profile...
September 19th 2016 12:45:01Verbose
Retrieved publishing profile: URI: app-staging.scm.ase-test1.p.azurewebsites.net:443  UserName: $WebApp__Staging
September 19th 2016 12:45:01Verbose
Using ID 'c6701672-0833-4e32-972a-f43d615fbe81' for connections to the remote server.
September 19th 2016 12:45:01Verbose
Pre-authenticating to remote agent URL 'https://app-staging.scm.ase-test1.p.azurewebsites.net/msdeploy.axd?site=WebApp(Staging)' as '$WebApp__Staging'.
September 19th 2016 12:45:01Error
Microsoft.Web.Deployment.DeploymentAgentUnavailableException: Could not complete the request to remote agent URL 'https://app-staging.scm.ase-test1.p.azurewebsites.net/msdeploy.axd?site=WebApp(Staging)'. ---> System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a send. ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
September 19th 2016 12:45:01Error
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
September 19th 2016 12:45:01Error
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   at System.Net.FixedSizeReader.ReadPacket(Byte[] buffer, Int32 offset, Int32 count)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
September 19th 2016 12:45:01Error
   at System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)
September 19th 2016 12:45:01Error
   at System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   at System.Net.ConnectStream.WriteHeaders(Boolean async)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at System.Net.HttpWebRequest.GetResponse()
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponseHelper(HttpWebRequest request)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponseHelper(HttpWebRequest request)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponse(HttpWebRequest request)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.PerformHeadRequestHelper(Boolean getVersionInfo, Version& maximumSupportedVersion, Version& minimumSupportedVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider..ctor(DeploymentProviderContext providerContext, DeploymentBaseContext baseContext, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentManager.CreateObjectPrivate(DeploymentProviderContext providerContext, DeploymentBaseOptions baseOptions, DeploymentObject sourceObject, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentManager.CreateDestinationObject(DeploymentProviderOptions providerOptions, DeploymentBaseOptions baseOptions, DeploymentObject sourceObject, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentObject.SyncTo(DeploymentProviderOptions providerOptions, DeploymentBaseOptions baseOptions, DeploymentSyncOptions syncOptions)
September 19th 2016 12:45:01Error
   at Calamari.Azure.Deployment.Conventions.AzureWebAppConvention.Install(RunningDeployment deployment) in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari.Azure\Deployment\Conventions\AzureWebAppConvention.cs:line 18
September 19th 2016 12:45:01Error
   at Calamari.Deployment.ConventionProcessor.RunInstallConventions() in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari\Deployment\ConventionProcessor.cs:line 60
September 19th 2016 12:45:01Error
   at Calamari.Deployment.ConventionProcessor.RunConventions() in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari\Deployment\ConventionProcessor.cs:line 28
September 19th 2016 12:45:01Error
Running rollback conventions...
September 19th 2016 12:45:01Error
Microsoft.Web.Deployment.DeploymentAgentUnavailableException: Could not complete the request to remote agent URL 'https://app-staging.scm.ase-test1.p.azurewebsites.net/msdeploy.axd?site=WebApp(Staging)'. ---> System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a send. ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
September 19th 2016 12:45:01Error
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
September 19th 2016 12:45:01Error
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   at System.Net.FixedSizeReader.ReadPacket(Byte[] buffer, Int32 offset, Int32 count)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
September 19th 2016 12:45:01Error
   at System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
September 19th 2016 12:45:01Error
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
September 19th 2016 12:45:01Error
   at System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)
September 19th 2016 12:45:01Error
   at System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)
September 19th 2016 12:45:01Error
   at System.Net.ConnectStream.WriteHeaders(Boolean async)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at System.Net.HttpWebRequest.GetResponse()
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponseHelper(HttpWebRequest request)
September 19th 2016 12:45:01Error
   --- End of inner exception stack trace ---
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponseHelper(HttpWebRequest request)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.GetHttpResponse(HttpWebRequest request)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider.PerformHeadRequestHelper(Boolean getVersionInfo, Version& maximumSupportedVersion, Version& minimumSupportedVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.AgentClientProvider..ctor(DeploymentProviderContext providerContext, DeploymentBaseContext baseContext, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentManager.CreateObjectPrivate(DeploymentProviderContext providerContext, DeploymentBaseOptions baseOptions, DeploymentObject sourceObject, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentManager.CreateDestinationObject(DeploymentProviderOptions providerOptions, DeploymentBaseOptions baseOptions, DeploymentObject sourceObject, String serverVersion)
September 19th 2016 12:45:01Error
   at Microsoft.Web.Deployment.DeploymentObject.SyncTo(DeploymentProviderOptions providerOptions, DeploymentBaseOptions baseOptions, DeploymentSyncOptions syncOptions)
September 19th 2016 12:45:01Error
   at Calamari.Azure.Deployment.Conventions.AzureWebAppConvention.Install(RunningDeployment deployment) in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari.Azure\Deployment\Conventions\AzureWebAppConvention.cs:line 18
September 19th 2016 12:45:01Error
   at Calamari.Deployment.ConventionProcessor.RunInstallConventions() in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari\Deployment\ConventionProcessor.cs:line 60
September 19th 2016 12:45:01Error
   at Calamari.Deployment.ConventionProcessor.RunConventions() in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari\Deployment\ConventionProcessor.cs:line 50
September 19th 2016 12:45:01Error
   at Calamari.Azure.Commands.DeployAzureWebCommand.Execute(String[] commandLineArguments) in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari.Azure\Commands\DeployAzureWebCommand.cs:line 86
September 19th 2016 12:45:01Error
   at Calamari.Program.Execute(String[] args) in Z:\BuildAgent\work\14ffc968155e4956\source\Calamari\Program.cs:line 38
September 19th 2016 12:45:02Fatal
The remote script failed with exit code 100
.....
